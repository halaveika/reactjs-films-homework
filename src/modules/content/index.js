import { contentReducer } from './contentReducer';

export default contentReducer;
