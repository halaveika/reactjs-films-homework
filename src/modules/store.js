import { createStore, applyMiddleware, combineReducers } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import contentReducer from './content';

const rootReducer = combineReducers({ content: contentReducer });

const middleware = [thunk];
const store = createStore(
  rootReducer,
  composeWithDevTools(
    applyMiddleware(...middleware),
  ),
);

export default store;
