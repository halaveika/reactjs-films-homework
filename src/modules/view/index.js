import { viewReducer } from './viewReducer';

export default viewReducer;
