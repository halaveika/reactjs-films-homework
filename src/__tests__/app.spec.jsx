import React from 'react';
import { create } from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { PersistGate } from 'redux-persist/integration/react';
import ShallowRenderer from 'react-test-renderer/shallow'
import App from '../app';
import { store,persistor} from '../modules/store';


window.matchMedia = window.matchMedia || function mMedia() {
  return {
    matches: false,
    addListener() {},
    removeListener() {},
  };
};

afterEach(() => {
  jest.clearAllMocks();
});

describe('test App component', () => {
  // it('should render App component', () => {
  //   const component = create(
  //     <Provider store={store}>
  //       <PersistGate loading={null} persistor={persistor}>
  //         <BrowserRouter>
  //           <App />
  //         </BrowserRouter>
  //       </PersistGate>
  //     </Provider>,
  //   );
  //   const tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });


  it('Matches snapshot ShallowRenderer', () => {
    const renderer = new ShallowRenderer()
    const result = renderer.render(
            <App />
    )
    expect(result).toMatchSnapshot()
  })
  
});



