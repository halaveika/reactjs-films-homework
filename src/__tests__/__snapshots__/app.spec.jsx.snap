// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test App component Matches snapshot ShallowRenderer 1`] = `
<React.Fragment>
  <Memo(Connect(Header)) />
  <AppRouter />
</React.Fragment>
`;
